mungegithub/mungers/rebuild.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"fmt"
+	"regexp"
+
+	"k8s.io/contrib/mungegithub/github"
+
+	"github.com/golang/glog"
+	githubapi "github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+)
+
+// RebuildMunger looks for situations where a someone has asked for an e2e rebuild, but hasn't provided
+// an issue
+type RebuildMunger struct{}
+
+var (
+	buildMatcher = regexp.MustCompile("@k8s-bot\\s+test\\s+this.*")
```

Comment:
`@k8s-bot e2e test this` and `@k8s-bot unit test this` also trigger the respective PR Jenkins jobs.